### ISSUE TITLE
[pandoc] binaries don't work on ARMv8

### ISSUE THREAD

**Issue by cstrouse** [chromebrew/chromebrew#3512]
At first I thought this was due to stripping behavior since it complains that it doesn't recognize the format but I think the real issue is that they only provide 64-bit binaries and while ARMv8 Chromebooks are 64-bit hardware CrOS is 32-bit.

```
chronos@localhost /usr/local/tmp $ pandoc
bash: /usr/local/bin/pandoc: cannot execute binary file: Exec format error
```

Once we fix `ghc` (see #3511) we can try creating a source build for `pandoc` and build 32-bit binaries for ARMv8.

**Comment by vincowl** [comment]
pandoc, ghc and cabal have been upgraded. No binary available at the moment, but does it solve the problem ?

**Comment by cstrouse** [comment]
No. The problem still exists. Also, this package needs to have stripping disabled.

**Comment by vincowl** [comment]
Ok, si it doesn't involve the ability to compile it from sources or not : it works when building from sources. Right ?

I don't know anything about Haskell so I'll leave it all to people more skilled about it.

What do you mean by disabling stripping ? Is it an arm need or a global one ? Works perfectly as is on my x86_64 machine.

**Comment by cstrouse** [comment]
The binaries are still 64-bit. Haskell stuff doesn't compile on here because `stack` only wants to install 64-bit versions of `cabal` and `ghc`.

**Comment by vincowl** [comment]
@cstrouse What's the output of `uname -m` please ?

**Comment by vincowl** [comment]
@cstrouse I have updated the `stack` package to take armv8 into account. I had a look at `ghc` but arm is not listed as a target. Since `ghc` is built using `stack`, I think the #3726 should do the trick if stack is compiled as 32 bit version. If it doesn't, may be we should add an equivalent to `-m32` flag to the compilation in case of armv8 archs.

**Comment by vincowl** [comment]
Maybe apply #3727 if errors are encountered with `cabal`

**Comment by cstrouse** [comment]
Someone is working on pandoc binaries for ARM (see https://github.com/jgm/pandoc/issues/5450)
